poseidon2/src/round_numbers.rs

```diff
@@ -1,24 +1,42 @@
+// As the security analysis of Poseidon2 is identical to that of Poseidon,
```

Comment:
This whole comment should be a file-level comment with `//!` rather than just `//`. (Also need `//!` at the start of the empty lines.)